tests/fault_tolerance/etcd_ha/test_vllm.py

```
@@ -443,3 +448,147 @@ def test_etcd_non_ha_shutdown_vllm_disaggregated(
+    t = threading.Thread(target=_run, daemon=True)
+    t.start()
+    return t
+
+
+def _freeze_vllm_engine_processes():
+    """SIGSTOP the vLLM engine (rank) subprocess(es); return frozen pids.
+
+    A frozen engine cannot finish or abort the in-flight request, so it stays
+    pinned in the push-endpoint inflight counter -- a non-cancellable inflight.
+    """
+    out = subprocess.run(["pgrep", "-f", "EngineCore"], capture_output=True, text=True)
```

Ruff (0.15.20)

[error] Starting a process with a partial executable path

(S607)